### ISSUE TITLE
isSubmitting gets set to true when the form is submitted, even when there are validation errors

### ISSUE THREAD

**Issue by scbrady** [jaredpalmer/formik#119]
When submit a form with formik, the `isSubmitting` prop is set to `true`, triggering a render and causing submit logic to fire in my view. This all works great unless there are validation errors. When I submit a form that has validation errors, `isSubmitting` is set to `true`, a render happens, then it is set to `false`. This causes weird glitches.

Is there a reason why `isSubmitting` is set to true, even when it isn't actually submitting? If not, I'd be happy to submit a PR.

**Comment by jaredpalmer** [comment]
Feel free to submit a PR

**Comment by jaredpalmer** [comment]
Right so `isSubmitting` is set to `true` because the user is trying to submit. It is set `false` if the form is found to be invalid. I guess it is somewhat confusing in the sense that `handleSubmit` may or may not be called. Not sure about the double render you are seeing. Can you give an example?

**Comment by scbrady** [comment]
Sure! I have a custom button:

```
<Button showsLoading submit isSubmitting={isSubmitting}>
```

That fires a redux action to make our page show a loading state as soon as `isSubmitting` is set to `true`. I then hide the loading state at the end of the `handleSubmit` function.

I've already worked around this problem in my code, but it just seemed like a bug and confused me for a while until I put in breakpoints to see what was happening. I just figured I'd check in to see if this was the intended behavior or not.
